src/protocol/vm/state_builder.rs

```diff
@@ -244,7 +244,7 @@ impl VMPoolStateBuilder {
             engine.state.init_account(
                 Address::parse_checksummed(to_checksum(token_address, None), None).map_err(
                     |_| {
-                        SimulationError::EncodingError(
+                        SimulationError::FatalError(
```

Comment:
Oh ok I just saw this now but I'd prefer if we would not deal with any checksummed addresses in the whole codebase. This was a mess in our cold codebase. Instead addresses should be just bytes and sometimes we represent them as a hex string - lower or uppercase should not matter.

EDIT: I realise outside of scope here again.